content/blog/20230731_upgrade-td-agent-v4-to-v5.md

```diff
@@ -0,0 +1,286 @@
+fluent-plugin-elasticsearch (5.3.0)
+fluent-plugin-flowcounter-simple (0.1.0)
+fluent-plugin-kafka (0.19.0)
+fluent-plugin-metrics-cmetrics (0.1.2)
+fluent-plugin-opensearch (1.1.0)
+fluent-plugin-prometheus (2.0.3)
+fluent-plugin-prometheus_pushgateway (0.1.0)
+fluent-plugin-record-modifier (2.1.1)
+fluent-plugin-rewrite-tag-filter (2.4.0)
+fluent-plugin-s3 (1.7.2)
+fluent-plugin-sd-dns (0.1.0)
+fluent-plugin-systemd (1.0.5)
+fluent-plugin-td (1.2.0)
+fluent-plugin-utmpx (0.5.0)
+fluent-plugin-webhdfs (1.5.0)
+```
+ 
+As for fluent-package v5, "fluent-plugin-concat" was installed under "/opt/fluent/lib/ruby/gems/3.2.0/gems/" directories.
+ 
+```
+$ ls -l /opt/fluent/lib/ruby/gems/3.2.0/gems/ |grep fluent-plugin*
+drwxr-xr-x.  5 root root  175  7月 14 03:14 fluent-plugin-calyptia-monitoring-0.1.3
+drwxr-xr-x.  4 root root  206  7月 14 03:16 fluent-plugin-concat-2.5.0
+drwxr-xr-x.  5 root root 4096  7月 14 03:14 fluent-plugin-elasticsearch-5.3.0
+drwxr-xr-x.  4 root root  205  7月 14 03:14 fluent-plugin-flowcounter-simple-0.1.0
+drwxr-xr-x.  6 root root  191  7月 14 03:14 fluent-plugin-kafka-0.19.0
+drwxr-xr-x.  5 root root  190  7月 14 03:14 fluent-plugin-metrics-cmetrics-0.1.2
+drwxr-xr-x.  5 root root 4096  7月 14 03:14 fluent-plugin-opensearch-1.1.0
+drwxr-xr-x.  5 root root  215  7月 14 03:14 fluent-plugin-prometheus-2.0.3
+drwxr-xr-x.  6 root root  238  7月 14 03:14 fluent-plugin-prometheus_pushgateway-0.1.0
+drwxr-xr-x.  4 root root  176  7月 14 03:14 fluent-plugin-record-modifier-2.1.1
+drwxr-xr-x.  3 root root  210  7月 14 03:14 fluent-plugin-rewrite-tag-filter-2.4.0
+drwxr-xr-x.  5 root root  230  7月 14 03:14 fluent-plugin-s3-1.7.2
+drwxr-xr-x.  3 root root  170  7月 14 03:14 fluent-plugin-sd-dns-0.1.0
+drwxr-xr-x.  3 root root   49  7月 14 03:14 fluent-plugin-systemd-1.0.5
+drwxr-xr-x.  5 root root  221  7月 14 03:14 fluent-plugin-td-1.2.0
+drwxr-xr-x.  5 root root  166  7月 14 03:14 fluent-plugin-utmpx-0.5.0
+drwxr-xr-x.  4 root root  191  7月 14 03:14 fluent-plugin-webhdfs-1.5.0
+```
+
+### 8. Start fluent-package v5 daemon.
+
+```
+$ sudo systemctl start fluent-package
+```
+
+### 9. Check if there are no error messages in fluentd logs.
+
+```
+$ tail -100f /var/log/fluent/fluentd.log
+```
+
+Now, upgrading steps are completed. Happy Logging!
+
+## Additional hints for v4 users
+
+### For Debian/Ubuntu
+
+* If you upgrade from v4, `td-agent` will be installed as a transitional package.
+  you can remove it safely with `sudo apt purge td-agent`.
+* `fluentd-apt-source` package will be marked as a transitional package.
+  you can remove it safely with `sudo apt purge fluentd-apt-source`.
+* If you want to enable `td-agent.service`, you must explicitly execute the following commands:
+
+  ```
+  $ sudo systemctl unmask td-agent
+  $ sudo systemctl enable fluentd
+  ```
+
+### For RHEL
+
+* If you upgrade from v4, `td-agent` will be replaced with `fluent-package`.
+  you can remove it safely with `sudo dnf remove td-agent`.
```

Comment:
Same concern as above.

Reply:
Oops, it is just wrong description. I'll fix it.

Reply:
Fixed.